cppcache/integration-test/fw_dunit.cpp

```diff
@@ -753,13 +756,22 @@ class TestSlave {
           DUNIT->setSlaveState(m_sId, SLAVE_STATE_TASK_ACTIVE);
           try {
             task->doTask();
+            if (task->m_isHeapAllocated) {
```

Comment:
Could we just make it a `std::shared_ptr` and let it do its magic?

Reply:
Not really. The framework uses sometimes creates instances in the heap and sometimes in the stack.